reccmp/tools/project.py

```diff
@@ -0,0 +1,457 @@
+        )
+
+    result += "\n"
+    result += textwrap.dedent(
+        """\
+        reccmp_configure()
+    """
+    )
+    return result
+
+
+def get_default_main_hpp(target_id: str) -> str:
+    return textwrap.dedent(
+        f"""\
+        #ifndef {target_id.upper()}_HPP
+        #define {target_id.upper()}_HPP
+
+        // VTABLE: {target_id} 0x10001000
+        // SIZE 0x8
+        class SomeClass {{
+            virtual ~SomeClass(); // vtable+0x00
+            int m_member;
+        }};
+
+        #endif /* {target_id.upper()}_HPP */
+        """
+    )
+
+
+def get_default_main_cpp(target_id: str, original_path: Path, hpp_path: Path) -> str:
+    target_type = executable_or_library(original_path)
+    match target_type:
+        case TargetType.EXECUTABLE:
+            entry_function = textwrap.dedent(
+                """\
+                #ifdef _WIN32
+                #include <windows.h>
+
+                // FUNCTION: {original_id} 0x10000020
+                int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PSTR lpCmdLine, int nCmdShow) {
+                    return 0;
+                }
+                #else
+                // FUNCTION: {original_id} 0x10000020
+                int main(int argc, char *argv[]) {
+                    return 0;
+                }
+                #endif
+            """
+            )
+        case TargetType.SHARED_LIBRARY:
+            entry_function = textwrap.dedent(
+                """\
+                #ifdef _WIN32
+                #include <windows.h>
+
+                // FUNCTION: {original_id} 0x10000020
+                BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved ) {
+                    return TRUE;
+                }
+                #endif
+            """
+            )
+    return textwrap.dedent(
+        f"""\
+        #include "{hpp_path.name}"
+
+        // FUNCTION: {target_id} 0x10000000
+        SomeClass::~SomeClass() {{
+        }}
+
+        // GLOBAL: {target_id} 0x10102000
+        // STRING: {target_id} 0x10101f00
+        const char* g_globalString = "A global string";
+
+        {entry_function}
+    """
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Project management", allow_abbrev=False
+    )
+    parser.set_defaults(action=None)
+    parser.add_argument(
+        "--version", action="version", version=f"%(prog)s {reccmp.VERSION}"
+    )
+    parser.add_argument(
+        "-C",
+        type=Path,
+        dest="cwd",
+        metavar="<path>",
+        default=Path.cwd(),
+        help="Run as if %(prog)s was started in %(metavar)s",
+    )
+    subparsers = parser.add_subparsers()
+
+    create_parser = subparsers.add_parser("create")
+    create_parser.set_defaults(action="CREATE")
+    create_parser.add_argument(
+        "--originals",
+        type=Path,
+        nargs="+",
+        metavar="ORIGINAL",
+        dest="create_originals",
+        required=True,
```

Comment:
According to the [documentation](https://docs.python.org/3/library/argparse.html#required), it is a best practice to not have required arguments starting with `-` or `--`. How about a syntax like ` reccmp-project create LEGO1.DLL` without `--originals`?

Reply:
I made these positional.